grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -109,4 +116,36 @@ class LocationDataService {
 
         return location
     }
+
+    Address createOrUpdateAddress(Map params, @Nullable String addressId) {
+        Address address
+        if (addressId) {
+            address = Address.findById(addressId)
+        }
+
+        if (!address) {
```

Comment:
You could just do:
```
if (!address) {
  address = new Address()
}
```
and then do the params assign code once

Reply:
done